src/core-services/account/mutations/addOrRemoveAccountGroups.js

```diff
@@ -0,0 +1,111 @@
+import SimpleSchema from "simpl-schema";
+import ReactionError from "@reactioncommerce/reaction-error";
+import AddOrRemoveAccountGroupsOperationType from "./AddOrRemoveAccountGroupsOperationType.js";
+
+const inputSchema = new SimpleSchema({
+  "accountId": String,
+  "userId": String,
+  "groups": {
+    type: Array, // groupIds that user belongs to
+    optional: true,
+    defaultValue: []
+  },
+  "groups.$": {
+    type: String
+  }
+});
+
+
+const getQueryAndOption = (updateGroupOperationType, groups) => {
+  let queryAndOptions = null;
+  switch (updateGroupOperationType) {
+    case AddOrRemoveAccountGroupsOperationType.ADD: {
+      queryAndOptions = {
+        query: {
+          $addToSet: {
+            groups: {
+              $each: groups
+            }
+          }
+        },
+        options: {
+          returnOriginal: false
+        }
+
+      };
+      break;
+    }
+    case AddOrRemoveAccountGroupsOperationType.DELETE: {
+      queryAndOptions = {
+        query: {
+          $pull: {
+            groups: {
+              $in: groups
+            }
+          }
+        },
+        options: {
+          multi: true
+        }
+      };
+      break;
+    }
+    default: return queryAndOptions;
+  }
+
+  return queryAndOptions;
+};
+/**
+ * @name accounts/addOrRemoveAccountGroups
+ * @memberof Mutations/Accounts
+ * @summary Adds or removes  groups on an account
+ * @param {Object} context - GraphQL execution context
+ * @param {Object} input - Necessary input for mutation. See SimpleSchema.
+ * @param {AddOrRemoveAccountGroupsOperationType} updateGroupOperationType - the type of operation to perform.
+ * @param {Object} input.groups - groups to append to
+ * @param {String} input.accountId - optional decoded ID of account on which entry should be updated, for admins
+ * @returns {Promise<Object>} with updated account
+ */
+export default async function addOrRemoveAccountGroups(context, input, updateGroupOperationType) {
```

Comment:
While this is still a draft, the GQL mutations must be discrete. A mutation resolver & mutation function must exist for both `add` and `remove`. Don't mix actions, or make mutations to 2 different actions with a switch. Even if it is redundant, and it does duplicate code.

Reply:
@mikemurray  ... Thanks ... will do that

Reply:
@mikemurray  .... Done